packages/fether-react/src/BackupPhrase/BackupPhrase.js

```diff
@@ -0,0 +1,394 @@
+// This file is part of Parity.
+//
+// SPDX-License-Identifier: BSD-3-Clause
+import * as CryptoJS from 'crypto-js';
+import { AccountCard, Header, Form as FetherForm } from 'fether-ui';
+import localForage from 'localforage';
+import { inject } from 'mobx-react';
+import React, { Component } from 'react';
+import { Link } from 'react-router-dom';
+
+import RequireHealthOverlay from '../RequireHealthOverlay';
+import withAccount from '../utils/withAccount';
+
+/*
+  feat #100: The user skipped phrase rewrite during account creation.
+  Then ->
+    Two cases:
+    1. The user hasn't confirmed recovery phrase backup yet.
+      -> In this case we show a warning and guide them to decrypt the phrase with their password and through to the rewrite step.
+    2. The user has confirmed backup of their recovery phrase, but just wants to view it once more.
+      -> In this case, we do not show warning, but we alow them to decrypt the phrase with their password.
+*/
+@inject('parityStore')
+@withAccount
+class BackupPhrase extends Component {
+  state = {
+    error: null,
+    password: '',
+    phrase: '',
+    phraseRewrite: '',
+    step: 1,
+    unlocked: false
+  };
+
+  componentDidMount () {
+    this.getPhrase();
+  }
+
+  handleChangePassword = ({ target: { value } }) => {
+    this.setState({ password: value });
+  };
+
+  // handle the user typed phrase rewrite
+  handleChangePhrase = ({ target: { value } }) => {
+    this.setState({ phraseRewrite: value });
+  };
+
+  handleSubmit = async event => {
+    const {
+      account: { address },
+      history
+    } = this.props;
+    const { password, phrase } = this.state;
+
+    // unset account flag
+    await localForage.removeItem(`__flagged_${address}`);
+
+    // AES encrypt the phrase
+    const encryptedPhrase = CryptoJS.AES.encrypt(phrase, password).toString();
+
+    // set flag as safe
+    await localForage.setItem(`__safe_${address}`, encryptedPhrase);
+
+    // go to account
+    history.push(`/tokens/${address}`);
+  };
+
+  // Get the actual phrase to rewrite
+  getPhrase = async () => {
+    const {
+      account: { address }
+    } = this.props;
+
+    const phrase =
+      (await localForage.getItem(`__flagged_${address}`)) ||
+      (await localForage.getItem(`__safe_${address}`));
+
+    // phrase is still encrypted
+    this.setState({ phrase });
+  };
+
+  onCopyOrPastePhrase = event => {
+    // Disable copying the phrase if in production.
+    // Keep it enabled for development.
+    if (process.env.NODE_ENV === 'production') {
+      event.preventDefault();
+      this.setState({
+        error:
+          'Copy and pasting is disabled for this step. Please type out your full recovery phrase.'
+      });
+
+      return false;
+    }
+  };
+
+  nextStep = event => {
+    const {
+      account: { address },
+      history,
+      location: { pathname }
+    } = this.props;
+
+    event.preventDefault();
+
+    const needsRewrite = pathname.split('/')[3] === 'true';
```

Comment:
can we use this.props.match.params.needsRewrite here? (in other places as well; might want to use a function)